google/cloud/examples/BUILD

```diff
@@ -41,15 +47,49 @@ cc_test(
     deps = [":gcs2cbt_lib"],
 )
 
+proto_library(
+    name = "hello_world_protos",
+    srcs = [
+        "hello_world_grpc/hello_world.proto",
+    ],
+    visibility = [
+        "//:__pkg__",
+    ],
+    deps = [],
+)
+
+cc_proto_library(
+    name = "hello_world_cc_proto",
+    visibility = [
+        "//:__pkg__",
+    ],
+    deps = [":hello_world_protos"],
+)
+
+cc_grpc_library(
+    name = "hello_world_cc_grpc",
+    srcs = [":hello_world_protos"],
+    grpc_only = True,
+    visibility = [
+        "//:__pkg__",
+    ],
+    deps = [":hello_world_cc_proto"],
+)
+
 cc_test(
     name = "grpc_credential_types",
     timeout = "long",
     srcs = ["grpc_credential_types.cc"],
+    copts = [
+        "-I$(GENDIR)",
```

Comment:
I'm surprised this `-I` is needed. I don't think we usually need that w/ proto deps. Do you know why it's needed in this case?

Reply:
But we do need something similar in our proto deps:

https://github.com/googleapis/google-cloud-cpp/blob/ebb6247af79d63df0e4ebc33c2cf18334a5223f2/bazel/googleapis.BUILD#L19-L29

and:

https://github.com/googleapis/google-cloud-cpp/blob/ebb6247af79d63df0e4ebc33c2cf18334a5223f2/google/cloud/BUILD#L100

Reply:
Oh, so this is to enable angle-bracket includes? If we could accomplish this similar to how `googleapis_system_includes` works using `includes = [ "$(GENDIR)" ]` on the proto/grpc targets instead of the users of that target, that may be preferable. But I'll let you see what works and pick one.

Reply:
Neither rule supports that parameter:

https://github.com/grpc/grpc/blob/cc9326c93650f5e853fe02922ddb476af70ce303/bazel/cc_grpc_library.bzl#L29
https://docs.bazel.build/versions/master/be/c-cpp.html#cc_proto_library

I could add an empty `cc_library()` rule with just that definition, hardly seems better, do you think it would be? I like :shrug: no idea :shrug: